### PR TITLE
calendar

### PR DESCRIPTION
**Pull request by NikitaKarbazin**
- [DEMO LINK](https://NikitaKarbazin.github.io/layout_calendar/)

_Add a number (Arial 30px) in the center of each day using ::before and [@for](https://sass-lang.com/documentation/at-rules/control/for)_ 
I made this without for loop, because i find another solution(and dont understand where put @for loop)

### CODE REVIEW

**Review by Kaunin1 — APPROVED** [review]

**Comment by Kaunin1** [comment]
Good job ! calendar functions  well 
code looks fine

File: src/styles/main.scss [hunk 1]
```diff
@@ -1,3 +1,91 @@
...
+    }
+
+    @each $day, $int in $days {
+      &--start-day_#{$day} {
+        .calendar__day:nth-child(1) {
+          margin-left: shiftSpace($shift, $int);
+        }
+      }
+    }
+    @for $day-is from 27 through 31 {
```
Comment (illiarerun) [inline src/styles/main.scss]: are there any months with 27 days? I don't think so
image

File: src/styles/main.scss [hunk 2]
```diff
@@ -1,3 +1,91 @@
+$gap-size: 1px;
+$calendar-margin: auto;
+$font-size: 30px;
+$border-size: 1px;
+$wigth-cell: 100px;
+$height-cell: 100px;
```
Comment (illiarerun) [inline src/styles/main.scss]: you don't need 2 variables with the same values
```suggestion
$cell-size 100px;
```

File: src/styles/main.scss [hunk 3]
```diff
@@ -1,3 +1,91 @@
...
+$width-calendar: $wigth-cell * 7 + $gap-size * 10;
+$day-count: 31;
+$shift: $wigth-cell + $gap-size;
+$days: (
+  "mon": 1,
+  "tue": 2,
+  "wed": 3,
+  "thu": 4,
+  "fri": 5,
+  "sat": 6,
```
Comment (illiarerun) [inline src/styles/main.scss]: put it into `_variables.scss`

File: src/styles/main.scss [hunk 4]
```diff
@@ -1,3 +1,91 @@
...
+  "sun": 7);
+
+@function shiftSpace($shift, $int_day) {
+  @return ($shift * $int-day) - $shift;
+}
```
Comment (illiarerun) [inline src/styles/main.scss]: `_functions.scss`

File: src/styles/main.scss [hunk 5]
```diff
@@ -1,3 +1,91 @@
...
+
+@mixin figure($width, $height) {
+  width: $width;
+  height: $height;
+}
```
Comment (illiarerun) [inline src/styles/main.scss]: `_mixins.scss`

File: src/styles/main.scss [hunk 6]
```diff
@@ -1,3 +1,91 @@
...
+
+html {
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: $font-size;
+}
+
 body {
   margin: 0;
```
Comment (illiarerun) [inline src/styles/main.scss]: `reset.scss`

File: src/styles/main.scss [hunk 7]
```diff
@@ -1,3 +1,91 @@
...
+      @include figure($wigth-cell, $height-cell);
+      background-color: #eee;
+      border: $border-size solid #000;
+      transition: 0.5s;
+
+      &:hover {
+        background-color: pink;
+        cursor: pointer;
+        transform: translateY(-20px);
+        transition: 0.5s;
```
Comment (illiarerun) [inline src/styles/main.scss]: redundant, you already have it for main class
```suggestion
```

File: src/styles/main.scss [hunk 8]
```diff
@@ -1,3 +1,91 @@
...
+    &__day {
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      text-align: center;
+      box-sizing: border-box;
+      @include figure($wigth-cell, $height-cell);
+      background-color: #eee;
+      border: $border-size solid #000;
+      transition: 0.5s;
```
Comment (illiarerun) [inline src/styles/main.scss]: you should specify properties here

File: src/styles/main.scss [hunk 9]
```diff
@@ -1,3 +1,91 @@
...
+        }
+      }
+    }
+
+    &__day {
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      text-align: center;
+      box-sizing: border-box;
```
Comment (illiarerun) [inline src/styles/main.scss]: use `*` selector for `box-sizing` since it should be the same for all elements

**Review by illiarerun — CHANGES_REQUESTED** [review]
GJ, now let's work on some fixes:

File: src/styles/main.scss [hunk 10]
```diff
@@ -1,3 +1,67 @@
...
+    text-align: center;
+    @include figure($cell-size);
+    background-color: #eee;
+    border: $border-size solid #000;
+
+    &:hover {
+      background-color: pink;
+      cursor: pointer;
+      transform: translateY(-20px);
+      transition: 0.5s;
```
Comment (illiarerun) [inline src/styles/main.scss:58]:
```suggestion
    border: $border-size solid #000;
   transition: background-color 0.5s, transform 0.5s;

    &:hover {
      background-color: pink;
      cursor: pointer;
      transform: translateY(-20px);
```

**Review by illiarerun — APPROVED** [review]
GJ, left one comment

**Pull request closed without merging**